gobblin-metrics/src/main/java/gobblin/metrics/event/GenericEvent.java

```diff
@@ -0,0 +1,23 @@
+/*
+ * Copyright (C) 2014-2015 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use
+ * this file except in compliance with the License. You may obtain a copy of the
+ * License at  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed
+ * under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+ * CONDITIONS OF ANY KIND, either express or implied.
+ */
+
+package gobblin.metrics.event;
+
+import java.util.Map;
+
+/**
+ * A generic event with a name and metadata.
+ */
+public abstract class GenericEvent {
```

Comment:
Both event name and metadata are owned by the sub classes. If callers use eventSubmitter.submit(eventName, metadata) then the eventName need to be made public. I prefer sla eventName to be private to the SlaEvent class